### PR TITLE
feat: add filter styling for search and resources page

### PR DESCRIPTION
**Pull request by WillKocur**
Adds styling to filters on search and resources page.

Note the filters on search are different and not in the designs so I've just touched them up to be more similar to the resources page.

https://unep-wcmc.codebasehq.com/projects/bip-website/tickets/8

### CODE REVIEW

**Comment by WillKocur** [comment]
@nym1k Discussed changes made :)

File: app/assets/stylesheets/components/filters.scss [hunk 1]
```diff
@@ -1,59 +1,82 @@
...
+      @include font(filter-label-small);
+    }
+  }
+
+  @include e(filter-toggle) {
+    @include flexbox;
+    @include align-items(center);
+    @include justify-content(center);
+    @include flex-shrink(0);
+    border: solid 2px $border--checkbox;
```
Comment (nym1k) [inline app/assets/stylesheets/components/filters.scss:58]: Should we change from px to rem?

File: app/assets/stylesheets/components/filters.scss [hunk 2]
```diff
@@ -1,59 +1,82 @@
...
+    margin-right: vr(2);
+    width: 2.25rem;
+    height: 2.25rem;
+
+    .search-page__filters & {
+      border-width: 1px;
```
Comment (nym1k) [inline app/assets/stylesheets/components/filters.scss:64]: Should we change from px to rem?

**Review by nym1k — CHANGES_REQUESTED** [review]
Just some minor changes

Also, when you hover over a ticked filter, the tick itself seems to have an underline too
Screenshot 2022-11-18 at 11 18 30

File: app/assets/stylesheets/components/filters.scss [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
-  @include e(body) {
-    padding: vr(1.25) 2px;
-  }
 
-  @include e(subtitle) {
-    @include font(fact-subtitle);
-    margin: 0;
-    line-height: vr(2.8333333);
+  @include e(bucket) {
+    padding: vr(1) 0;
```
[the remark this answers is not in the document]
Comment (WillKocur) [reply app/assets/stylesheets/components/filters.scss]: Ah yes, I think this was for the filters in the dropdown on search, but have removed padding and put margin on the dropdown trigger

**Comment by WillKocur** [comment]
> Just some minor changes
> 
> Also, when you hover over a ticked filter, the tick itself seems to have an underline too Screenshot 2022-11-18 at 11 18 30

Fixed using link-reset, but can address this re your comment in the other PR if needed :)

File: app/assets/stylesheets/components/filters.scss [hunk 4]
```diff
@@ -1,59 +1,83 @@
...
+    @include align-items(center);
+    @include justify-content(center);
+    @include flex-shrink(0);
+    border: solid 0.125rem $border--checkbox;
+    margin-right: vr(2);
+    width: 2.25rem;
+    height: 2.25rem;
+
+    .search-page__filters & {
+      border-width: 1px;
```
Comment (nym1k) [inline app/assets/stylesheets/components/filters.scss:65]: Not sure if the px here were missed for this border

**Review by nym1k — CHANGES_REQUESTED** [review]
One tiny change then ready to merge

**Comment by WillKocur** [comment]
Ah missed that, yes! Done now @nym1k

**Review by nym1k — APPROVED** [review]

**Pull request merged**